image_data_guide/01_download_data.ipynb

```diff
@@ -0,0 +1,546 @@
+      "[Parallel(n_jobs=5)]: Done 118 tasks      | elapsed:   10.8s\n",
+      "[Parallel(n_jobs=5)]: Done 278 tasks      | elapsed:   25.1s\n",
+      "[Parallel(n_jobs=5)]: Done 502 tasks      | elapsed:   44.0s\n",
+      "[Parallel(n_jobs=5)]: Done 790 tasks      | elapsed:  1.2min\n",
+      "[Parallel(n_jobs=5)]: Done 1142 tasks      | elapsed:  1.7min\n",
+      "[Parallel(n_jobs=5)]: Done 1558 tasks      | elapsed:  2.4min\n",
+      "[Parallel(n_jobs=5)]: Done 2038 tasks      | elapsed:  3.2min\n",
+      "[Parallel(n_jobs=5)]: Done 2500 out of 2500 | elapsed:  3.9min finished\n"
+     ]
+    }
+   ],
+   "source": [
+    "with parallel_backend('threading', n_jobs=5):\n",
+    "    Parallel(verbose=3)(delayed(download_image)(a['coco_url'], sample_dir) for a in sample_annos.values())"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<pre>\n",
+    "</pre>"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Combine with CIFAR-10 Frog Data\n",
+    "The COCO dataset doesn't include any images of frogs, but let's say our model must also be able to label images of frogs. To fix this we can download another dataset of images which includes frogs, sample 250 frog images and add them to our existing image data. These images are much smaller (32x32) so they will appear pixelated and blurry when we increase the size of them to (244x244). We'll use the CIFAR-10 dataset to achieve this. As you'll see the CIFAR-10 dataset comes formatted in a very different manner from COCO dataset. We must process the CIFAR-10 data into individual image files so that it's congruent to our COCO images.\n",
+    "___"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Download and Extract the CIFAR-10 Dataset"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 18,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "--2020-10-05 05:23:37--  https://www.cs.toronto.edu/%7Ekriz/cifar-10-python.tar.gz\n",
+      "Resolving www.cs.toronto.edu (www.cs.toronto.edu)... 128.100.3.30\n",
+      "Connecting to www.cs.toronto.edu (www.cs.toronto.edu)|128.100.3.30|:443... connected.\n",
+      "HTTP request sent, awaiting response... 200 OK\n",
+      "Length: 170498071 (163M) [application/x-gzip]\n",
+      "Saving to: ‘cifar-10-python.tar.gz.2’\n",
+      "\n",
+      "cifar-10-python.tar 100%[===================>] 162.60M  20.6MB/s    in 8.3s    \n",
+      "\n",
+      "2020-10-05 05:23:46 (19.6 MB/s) - ‘cifar-10-python.tar.gz.2’ saved [170498071/170498071]\n",
+      "\n"
+     ]
+    }
+   ],
+   "source": [
+    "!wget https://www.cs.toronto.edu/%7Ekriz/cifar-10-python.tar.gz"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 19,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "tf = tarfile.open('cifar-10-python.tar.gz')\n",
```

Comment:
How large is the EBS volume for your NI and how large is this file? Re: I got an: 

`OSError: [Errno 12] Cannot allocate memory`

Reply:
I checked my volume and had room... seem like the unzip process it eating too much RAM.

Reply:
note added to the Downloading Data notebook to use a larger instance size if experiencing network or out of memory errors